src/features/world/roomMachine.ts

```diff
@@ -8,27 +8,24 @@ import { Bumpkin } from "features/game/types/game";
 import { INITIAL_BUMPKIN } from "features/game/lib/constants";
 import { BumpkinParts } from "lib/utils/tokenUriBuilder";
 
-type RoomSchema = any;
+export type Rooms = {
+  plaza: Room<PlazaRoomState> | undefined;
+  auction_house: Room<PlazaRoomState> | undefined;
+  clothes_shop: Room<PlazaRoomState> | undefined;
+};
+export type RoomId = keyof Rooms;
 
-type RoomId = "plaza" | "auction_house";
 export interface ChatContext {
   jwt: string;
   farmId: number;
   bumpkin: Bumpkin;
-  room?: Room<RoomSchema>;
```

Comment:
I have changed `room` to `rooms` to track state across multiple scenes independently.